### PR TITLE
Make objects pickleable

### PR DESCRIPTION
**Pull request by eunwoosh**
### Summary
This PR makes some objects pickleable. This is for preparing following HPO PR.
Except the change related to make pytorchcv model, others just change code structure so it doesn't change which code is executed.
Changes are as below:

- Code to deal with preprocessor error is moved to each file per tasks.
- `__reduce__()` is added in `MemCacheHandler`.
- `__reduce__()` is added in `OTXDataModule`.
- way to make pytorchcv model for detection is changed.

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Review by vinnamkim — APPROVED** [review]

File: src/otx/core/model/entity/utils/__init__.py
Comment (harimkang) [inline src/otx/core/model/entity/utils/__init__.py:1]: This folder is named `utils`, but I'm not sure if that's the right name. 
I don't know if the name like `utils.mmaction.create_model` is appropriate, but I can't think of anything off the top of my head, so feel free to ignore it.

File: src/otx/core/model/entity/utils/mmaction.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        """Which device being used."""
+        try:
+            buf = next(self.buffers())
+        except StopIteration:
+            return super().device
+        else:
+            return buf.device
+
+
+def create_model(config: DictConfig, load_from: str | None) -> tuple[nn.Module, dict[str, dict[str, int]]]:
```
Comment (harimkang) [inline src/otx/core/model/entity/utils/mmaction.py:44]: The reason these functions can't be combined into one is because of mmX's import structure, right?

**Review by harimkang — COMMENTED** [review]
Overall LGTM. I left a few comments.

**Review by sungmanc — APPROVED** [review]

File: src/otx/core/model/entity/utils/__init__.py
Comment (eunwoosh) [reply src/otx/core/model/entity/utils/__init__.py:1]: yeah I just thought that it's utils for creating a model. Maybe we will be able to move it when we find better place.

File: src/otx/core/model/entity/utils/mmaction.py (hunk 1 above)
Comment (eunwoosh) [reply src/otx/core/model/entity/utils/mmaction.py:44]: you're right. Each framework needs to use own `MODELS` registry.

**Review by harimkang — APPROVED** [review]

**Pull request merged**